### DISCUSSION TITLE
Is Nuxt3 still hybrid?

### DISCUSSION THREAD

**Discussion by Asxer** [nuxt/nuxt#16557] [Questions]
Hello! I can not find a way to fetch data like it was done in Nuxt2 when I had a `fetch` hook that is called on SSR when I refresh the page and on a client when I just navigate there on the client side. All that I can find is `useFetch` composable but it is just using a cropped version of Axios(I mean without Rest methods like Put, Patch, Delete, without interceptors, etc). I thought useAsyncData is something like what I needed but this code in the setup() method
```
useAsyncData(async () => await settingsStore.fetchSettings());
```
is not called at all. So I'm confused. If I need to fetch data in beforeMount hook why do I even need Nuxt? Where is the core function that made Nuxt helpful?

**Comment by TheAlexLichter** [comment]
Hey! Nuxt.js definitely still supports querying data on the server-side for the initial request and on subsequent requests on the client-side (if needed).

`useFetch` uses `useAsyncData` under the hood. Both composables are very suitable for this use case, but please be aware that you should `await` them in `<script setup>`.

Please see e.g. [this example](https://stackblitz.com/github/nuxt/framework/tree/main/examples/composables/use-fetch?file=app.vue&theme=dark)

**Reply by Asxer** [reply]
Adding await into the setup script helped, thank you! It looks like now the setup method works instead of the fetch method in Nuxt2. It is the thing that I would prefer to find in documentation to better understand what's going on now. 
Then I met a strange bug and I spend a whole day to get what was going on. Maybe the problem that I found could help somebody. 
The core of my issue was in that plugin of Axios(where init all interceptors for pending requests count and auth tokens) was called BEFORE the `persistedState` plugin(from this package: https://github.com/prazdevs/pinia-plugin-persistedstate). 
That's why there was nothing in the store, I got an error in the console of server rendering, didn't see it in the browser, and thought that server rendering just don't work. 
It looks like plugins are called in _alphabetical order_. I've renamed plugins/persistedState.ts to plugins/_persistedState.ts to ensure that it was called before all other plugins and now everything works properly!

**Reply by TheAlexLichter** [reply]
Have you checked out the [Nuxt 2 to Nuxt 3 migration guide](https://v3.nuxtjs.org/migration/component-options)? 
If something is missing, please feel free to raise an issue or PR ☺️

Regarding auto importing plugins, this is also denoted [in the docs](https://v3.nuxtjs.org/guide/directory-structure/plugins#plugin-registration-order)

**Reply by Asxer** [reply]
How could I find this page? This is not presented in the Nuxt 3 documentation navigation and when I tried to find it in google I'm always getting to [this page](https://v3.nuxtjs.org/migration/overview). It looks like I'm over-tired from all of it because only after our comment I figured out that after this page there were other pages.

What about plugins the main confusion wasn't in order indeed but in that the store did not work as expected, and reordering was only a solution.

I've spent over a week now initiating the project the way I want, fighting with the framework, and getting started to work, but in Nuxt 3, absolutely nothing works as expected. I'm close to giving up and redoing everything to Nuxt 2, because there are no damn Proxi, Pinia, Composition API, etc. Sorry for this, I am just completely frustrated. 

Now, I'm fighting with another problem with navigation, for some reason using the store inside method ruined navigation with a completely unclear error message but it is another question.

**Reply by Asxer** [reply]
Also, I found that construction

```
await useAsyncData('settings', async () => await settings.fetchSettings());
```

is not working either. The only construction that works properly is  

```
await settings.fetchSettings()
```

Otherwise, the function is just not called. I don't want to keep all data in components, I need to store it in the right way to have one source of truth - in store.

**Reply by TheAlexLichter** [reply]
> How could I find this page?

Get started > Upgrade guide > Nuxt 2 to Nuxt 3

> I've spent over a week now initiating the project the way I want, fighting with the framework, and getting started to work, but in Nuxt 3, absolutely nothing works as expected. I'm close to giving up and redoing everything to Nuxt 2, because there are no damn Proxi, Pinia, Composition API, etc. Sorry for this, I am just completely frustrated.

* Proxy will be supported by nitro by default, until then you can use e.g. https://github.com/Teranode/nuxt-module-alternatives/tree/master/%40nuxtjs-alt/proxy
* Composition API works as intended
* Pinia is working fine with the pinia Nuxt module

> Now, I'm fighting with another problem with navigation, for some reason using the store inside method ruined navigation with a completely unclear error message but it is another question.

That's hard to debug without actual code 🙈
